store/root/store.go

```diff
@@ -216,46 +220,85 @@ func (s *Store) Commit() ([]byte, error) {
 	}
 
 	// commit SC
-	commitInfo, err := s.commitSC(version, changeSet)
-	if err != nil {
+	if err := s.commitSC(); err != nil {
 		return nil, fmt.Errorf("failed to commit SC stores: %w", err)
 	}
 
-	s.lastCommitInfo = commitInfo
-	s.lastCommitInfo.Timestamp = s.commitHeader.GetTime()
+	if s.commitHeader != nil {
+		s.lastCommitInfo.Timestamp = s.commitHeader.GetTime()
+	}
 
 	s.rootKVStore.Reset()
+	s.workingHash = nil
 
-	return s.lastCommitInfo.Hash(), nil
+	return s.WorkingHash()
```

Comment:
it looks weird, @line 232 it sets nil and call again?

Reply:
Good catch!